definitions/features/katello_service.rb

```diff
@@ -27,13 +29,67 @@ def make_start(spinner, options = {})
       spinner.update 'No katello service to start'
     else
       filters = "--only #{services.join(',')}"
-      spinner.update 'Starting the katello services..'
+      spinner.update 'Starting katello services..'
       execute!("katello-service start #{filters}")
     end
   end
 
+  def restart(spinner, options = {})
+    spinner.update 'Restarting katello services..'
+    if options[:only] || options[:exclude]
+      filters = construct_filters_for_restart(options)
+      execute!("katello-service restart #{filters}")
+    else
+      execute!('katello-service restart')
+    end
+  end
+
+  def run_with_hammer_ping_retry(spinner, action = '', services_options = {})
+    meth_call_by_action(action, spinner, services_options)
+    spinner.update 'Sometimes, services will take time to be ready'
+    begin
+      Timeout.timeout(30) do
+        retry_hammer_ping(spinner)
+      end
+    rescue Timeout::Error => e
+      logger.error e.message
+    end
+  end
+
   private
 
+  def retry_hammer_ping(spinner)
+    RETRIES_FOR_SERVICES_RESTART.times do |retry_count|
+      spinner.update "Retry #{retry_count + 1}: checking status by hammer ping"
+      result = feature(:hammer).hammer_ping_cmd
+      if result[:success]
+        spinner.update 'All services are running.'
+        break
+      elsif retry_count < (RETRIES_FOR_SERVICES_RESTART - 1)
+        spinner.update 'Lets try checking status once again'
+        sleep 2
```

Comment:
the 2 should be extracted to a constant. Also, perhaps something like 30 s would be more suitable interval

Reply:
@iNecas , 
We don't need `Timeout` once we apply `30 s sleep`. I have added it so that it will only wait max up to 30 s.
I will remove it and add 30 s sleep.